packages/hardhat/test/Quest.test.ts

```diff
@@ -64,165 +84,262 @@ describe("[Contract] Quest", function () {
           .withArgs(evidence, player.address, claimAmount);
       });
 
-      it("should revert if the claim is greater than the quest funds", async function () {
+      it("SHOULD revert if the claim is greater than the quest funds", async function () {
         // Arrange
-        const claimAmount = questFunds + 1;
+        const claimAmount = questFunds.add(fromNumber(1));
         const evidence = hashToBytes("evidence1");
         const quest = await deployQuest(
           "fakeTitle",
           "0x",
           rewardToken,
           epoch0,
           govern.address,
-          founder.address,
-          questFunds
+          creator.address,
+          questFunds,
+          depositToken,
+          depositAmount,
+          creator
         );
 
         // Act
         const act = () =>
-          quest.claim(evidence, player.address, claimAmount, false);
+          quest
+            .connect(govern)
+            .claim(evidence, player.address, claimAmount, false);
 
         // Assert
         await expect(act()).to.be.revertedWith(
           "ERC20: transfer amount exceeds balance"
         );
       });
 
-      it("should revert if the there is no provided evidence", async function () {
+      it("SHOULD revert WHEN the there is no provided evidence", async function () {
         // Arrange
-        const claimAmount = questFunds + 1;
+        const claimAmount = questFunds.add(fromNumber(1));
         const evidence = [];
         const quest = await deployQuest(
           "fakeTitle",
           "0x",
           rewardToken,
           epoch0,
           govern.address,
-          founder.address,
-          questFunds
+          creator.address,
+          questFunds,
+          depositToken,
+          depositAmount,
+          creator
         );
 
         // Act
         const act = () =>
-          quest.claim(evidence, player.address, claimAmount, false);
+          quest
+            .connect(govern)
+            .claim(evidence, player.address, claimAmount, false);
 
         // Assert
         await expect(act()).to.be.revertedWith("ERROR: No evidence");
       });
     });
 
-    it("should revert if caller is not govern", async function () {
+    it("SHOULD revert WHEN caller is not govern", async function () {
       // Arrange
-      const rewardToken = await ethers.getContract("TokenMock");
       const quest = await deployQuest(
         "fakeTitle",
         "0x",
         rewardToken,
         epoch0,
         govern.address,
-        founder.address,
-        0
+        creator.address,
+        fromNumber(0),
+        depositToken,
+        depositAmount,
+        creator
       );
 
       // Act
       const act = () =>
         quest
           .connect(player)
-          .claim(hashToBytes("evidence1"), player.address, 0, true); // player claims by himself (should throw)
+          .claim(hashToBytes("evidence1"), player.address, fromNumber(0), true); // player claims by himself
```

Comment:
Oh! So just Govern claim it? When? Some schedule?

Reply:
Yeah the idea behind that is to force a timer of (by exemple) 7 days to let people challenge it

Reply:
And govern have a delay execution mecanism